dapp/app/ts/library/conversion.ts

```diff
@@ -29,3 +29,9 @@ export function daiHrdToDai(attodaiHrd: bigint, attodaiPerDaiHrd: bigint, rontod
 	const elapsed = nowSeconds - startTimeSeconds
 	return (Number(rontodsr) / 10**27) ** elapsed * Number(attodaiHrd) / 10**18 * Number(attodaiPerDaiHrd) / 10**18
 }
+
+export function tryHexStringToBigint(hex: string): bigint | undefined {
+	const match = /^(?:0x)?([a-fA-F0-9]{40})$/.exec(hex)
```

Comment:
this is really the best way to do it?

Reply:
I'm open to other options, but this seemed like the simplest way to identify whether a string is a valid address.

🤔 I should rename that fucntion to `HexStringAddressToBigint`...

Reply:
Marked as resolved because I renamed the function.  If you have ideas on how better to test if a provided string is an address let me know.  We _could_ do checksum validation, but we don't currently have a good error reporting system so if validation fails the user would likely be confused if they grabbed a non-checksummed address.

Reply:
i guess i'm just surprised we still have to regexp to match hex strings for conversion. checksum, not mvp.

Reply:
Actually, checksum validation is an async method and React isn't a big fan of async methods so I would need some scaffolding to hook it all up.  That makes me want to punt on checksum validation for now.

Reply:
I do have functions I could use instead, but they throw and I specifically want this method to return `undefined` on failure rather than throw an error.
```typescript
Bytes.fromHexString(destinationString).toUnsignedBigint()
```

Reply:
(when it returns `undefined` the submit button is hidden and the input field gets a red highlight)

Reply:
so regexp or flow control in try catch, gotcha.